// onconnect-movies-api/src/index.js
var util = require('./util');

module.exports = {
    init: function(key, secret) {
        this.apiKey = key;
        this.secret = secret;
    },
    getMoviesByLocation: function(lat, long, cb) {
        var today = new Date();
        var dd = today.getDate();
        var mm = today.getMonth() + 1;
        var yyyy = today.getFullYear();
        
        util.makeRequest({
                uri: 'movies/showings',
                startDate: yyyy+'-'+mm+'-'+dd,
                lat: lat,
                lng: long
            }, {
                apiKey: this.apiKey || '',
                secret: this.secret || ''
            },
            function(res) {
                if (res.error) {
                    cb(res);
                    return;
                }

                cb(res.map(function(movie) {
                    var showtimes = movie.showtimes.map(function(showtime) {
                        return {
                            theatreId: showtime.theatre.id,
                            theatreName: showtime.theatre.name,
                            dateTime: showtime.dateTime
                        }
                    })
                    return {
                        id: movie.tmsId,
                        title: movie.title,
                        showtimes: showtimes,
                        releaseDate: movie.releaseDate,
                        genres: movie.genres,
                        description: movie.longDescription,
                        topCast: movie.topCast,
                        directors: movie.directors,
                        officialUrl: movie.officialUrl,
                        preferredImage: movie.preferredImage,
                        qualityRating: movie.qualityRating
                    }
                }))
            })
    }
};
